demos/sd-turbo/index.js

```diff
@@ -942,6 +953,23 @@ const ui = async () => {
             break;
     }
 
+    if (config.devicetype.toLowerCase().indexOf("cpu") > -1 || config.provider.toLowerCase().indexOf("wasm") > -1) {
+        device.innerHTML = "CPU";
+        badge.setAttribute("class", "cpu");
+        document.body.setAttribute("class", "cpu");
+    } else if (
+        config.devicetype.toLowerCase().indexOf("gpu") > -1 ||
+        config.provider.toLowerCase().indexOf("webgpu") > -1
+    ) {
+        device.innerHTML = "GPU";
+        badge.setAttribute("class", "");
+        document.body.setAttribute("class", "gpu");
+    } else if (config.devicetype.toLowerCase().indexOf("npu") > -1) {
```

Comment:
Repeating `config.devicetype.toLowerCase()` 3 times feels much, rather than assigning to a variable once, but it's minor.